llvm/lib/Target/RISCV/RISCVMacroFusion.cpp

```diff
@@ -1,210 +0,0 @@
-//===- RISCVMacroFusion.cpp - RISC-V Macro Fusion -------------------------===//
```

Comment:
Do I need to migrate all my downstream macrofusions to tblgen now?

Reply:
No, you don't need to.
You can keep this file and override `getMacroFusions()` in `RISCVSubtarget`:
```cpp
  std::vector<MacroFusionPredTy> getMacroFusions() const override {
    std::vector<MacroFusionPredTy> Fusions =
        RISCVGenSubtargetInfo::getMacroFusions();
    Fusions.push_back(shouldScheduleAdjacent);
    return Fusions;
  }
```
`shouldScheduleAdjacent` is the predicator for your downstream macrofusions.